src/corehost/src/pal.unix.cpp

```diff
@@ -151,9 +151,9 @@ bool pal::file_exists(const pal::string_t& path)
     return (::stat(path.c_str(), &buffer) == 0);
 }
 
-std::vector<pal::string_t> pal::readdir(const pal::string_t& path)
+void pal::readdir(const pal::string_t& path, std::vector<pal::string_t>* list)
 {
-    std::vector<pal::string_t> files;
+    std::vector<pal::string_t>& files = *list;
```

Comment:
The function neither throws nor returns. This is a mandatory parameter, having it as NULL is pointless and I don't want to return an error code for this null check. An assert would be reasonable here. CLI doesn't do debug builds. I will trace log here and silently return if NULL.